idea/test-balance/src/services/genesis-hash/genesis-hash.service.ts

```diff
@@ -0,0 +1,41 @@
+import { KafkaMessage } from 'kafkajs';
+import { initLogger, JSONRPC_ERRORS, KAFKA_TOPICS } from '@gear-js/common';
+
+import { transferService } from '../transfer/transfer.service';
+import { gearService } from '../../gear';
+import { kafkaProducer } from '../../kafka/producer';
+
+const logger = initLogger('genesisHashService');
+
+export const genesisHashService = {
+  async sendByPayload(message: KafkaMessage, payload: any): Promise<void> {
+    let result;
+
+    try {
+      const isPossibleToTransfer = await transferService.isPossibleToTransfer(payload.address, payload.genesis);
+      if (isPossibleToTransfer) {
+        const transferBalance = await gearService.transferBalance(payload.address);
+        result = { result: transferBalance };
+      } else {
+        result = { error: JSONRPC_ERRORS.TransferLimitReached.name };
+      }
+    } catch (error) {
+      logger.error(error.message, error.stack);
+      result = { error: JSONRPC_ERRORS.InternalError.name };
+    }
+
+    await sendReply(message, JSON.stringify(result));
+  },
+  async sendApiGateway(): Promise<void> {
```

Comment:
`sendApiGateway`? What does it mean?

Reply:
Yeah, I see the main idea, but sendApiGateway sounds strange as if you are sending some apiGateway. Maybe `sendGenesis` would be better or smth else